### ISSUE TITLE
Raspberry B+ Segmentfault

### ISSUE THREAD

**Issue by tkndrm** [gotify/server#99]
Hi, cool project. I really want to use it on my (old) Raspberry B+ (arm6). Of course it doesn't work with the arm7 binary/docker version and give me _segmentfault_ when i run it. Is it possible to support arm6 and release a binary for it?

### German:
Hi, super Projekt. Würde es sehr gerne auf mein (alten) Raspberry B+ (arm6) laufen lassen. Natürlich klappt die arm7 binary/docker version nicht und gibt _Speicherzugriffsfehler_ zurück. Ist es möglich eine funktionieren binary zu releasen?

> $ uname -a
Linux technodrom 4.14.85+ #1173 Tue Dec 4 17:45:01 GMT 2018 armv6l GNU/Linux
>  $ ./gotify-linux-arm-7 
Speicherzugriffsfehler

**Comment by tkndrm** [comment]
Thanks, it output me this:
> $ ./gotify-linux-arm-6
> ./gotify-linux-arm-6: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.28' not found (required by ./gotify-linux-arm-6)

i try to fix it by installing `libc6` via `apt-get`. no difference.

**Comment by jmattheis** [comment]
Could you do a `ldd --version` and paste the output? I compiled Gotify with version 2.28 and your pi probably has an older version. See https://askubuntu.com/questions/421642/libc-so-6-version-glibc-2-14-not-found

**Comment by tkndrm** [comment]
Yes, looks like `GLIBC_2.24` is the latest on Raspberry B+  with Raspbian 9.6 (stretch)

> $ ldd --version
> ldd (Debian GLIBC 2.24-11+deb9u3) 2.24

>$ ldd gotify-linux-arm-6
> ./gotify-linux-arm-6: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.28' not found (required by ./gotify-linux-arm-6)
	/usr/lib/arm-linux-gnueabihf/libarmmem.so (0xb6ebf000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6eac000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb6e83000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb6d44000)
	/lib/ld-linux-armhf.so.3 (0xb6ed5000)

> $ strings /lib/arm-linux-gnueabihf/libc.so.6 | grep ^GLIBC_
GLIBC_2.4
GLIBC_2.5
GLIBC_2.6
GLIBC_2.7
GLIBC_2.8
GLIBC_2.9
GLIBC_2.10
GLIBC_2.11
GLIBC_2.12
GLIBC_2.13
GLIBC_2.14
GLIBC_2.15
GLIBC_2.16
GLIBC_2.17
GLIBC_2.18
GLIBC_2.22
GLIBC_2.23
GLIBC_2.24
GLIBC_PRIVATE

**Comment by jmattheis** [comment]
@tkndrm https://travis-ci.org/gotify/server/jobs/472423467#L936 travis-ci has the version 2.19 which also isn't listed in your strings output.

You could compile Gotify yourself either on your pi or a system with sufficient glibc version.

**Comment by jmattheis** [comment]
How to build gotify/server is described here: https://gotify.net/docs/dev-environment

Closing this issue for now.

**Comment by goofus** [comment]
I am having the same problem and frankly I can't get the compilation done. I am not that familiar with go, what I tried was:

```sh
$ git clone https://github.com/gotify/server
$ cd server
$ go get
go install: no install location for directory /home/gnorts/gotify/server outside GOPATH
        For more details see: go help gopath
```
and
```sh
$ go get github.com/gotify/server
# github.com/jinzhu/gorm
GO/src/github.com/jinzhu/gorm/main.go:25: undefined: sync.Map
# github.com/go-sql-driver/mysql
GO/src/github.com/go-sql-driver/mysql/connection.go:475: undefined: driver.TxOptions
GO/src/github.com/go-sql-driver/mysql/connection.go:481: undefined: sql.IsolationLevel
GO/src/github.com/go-sql-driver/mysql/connection.go:481: undefined: sql.LevelDefault
GO/src/github.com/go-sql-driver/mysql/connection.go:495: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/connection.go:514: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/connection.go:548: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/connection.go:567: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/connection.go:631: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/utils.go:730: undefined: driver.NamedValue
GO/src/github.com/go-sql-driver/mysql/utils.go:742: undefined: driver.IsolationLevel
GO/src/github.com/go-sql-driver/mysql/connection.go:481: too many errors
# github.com/lib/pq
GO/src/github.com/lib/pq/conn_go18.go:13: undefined: driver.NamedValue
GO/src/github.com/lib/pq/conn_go18.go:31: undefined: driver.NamedValue
GO/src/github.com/lib/pq/conn_go18.go:45: undefined: driver.TxOptions
GO/src/github.com/lib/pq/conn_go18.go:48: undefined: sql.IsolationLevel
GO/src/github.com/lib/pq/conn_go18.go:49: undefined: sql.LevelDefault
GO/src/github.com/lib/pq/conn_go18.go:51: undefined: sql.LevelReadUncommitted
GO/src/github.com/lib/pq/conn_go18.go:53: undefined: sql.LevelReadCommitted
GO/src/github.com/lib/pq/conn_go18.go:55: undefined: sql.LevelRepeatableRead
GO/src/github.com/lib/pq/conn_go18.go:57: undefined: sql.LevelSerializable
GO/src/github.com/lib/pq/notify.go:790: undefined: time.Until
GO/src/github.com/lib/pq/notify.go:790: too many errors
# github.com/gotify/server/api/stream
GO/src/github.com/gotify/server/api/stream/stream.go:166: u.Hostname undefined (type *url.URL has no field or method Hostname)
GO/src/github.com/gotify/server/api/stream/stream.go:171: u.Hostname undefined (type *url.URL has no field or method Hostname)
# github.com/rogpeppe/go-internal/module
GO/src/github.com/rogpeppe/go-internal/module/module.go:350: undefined: sort.Slice
# github.com/gobuffalo/syncx
GO/src/github.com/gobuffalo/syncx/byte_map.go:15: undefined: sync.Map
# github.com/markbates/oncer
GO/src/github.com/markbates/oncer/oncer.go:7: undefined: sync.Map
GO/src/github.com/markbates/oncer/oncer.go:18: undefined: sync.Map
# golang.org/x/crypto/acme/autocert
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:301: hello.SupportedProtos undefined (type *tls.ClientHelloInfo has no field or method SupportedProtos)
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:311: hello.SignatureSchemes undefined (type *tls.ClientHelloInfo has no field or method SignatureSchemes)
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:314: hello.SignatureSchemes undefined (type *tls.ClientHelloInfo has no field or method SignatureSchemes)
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:315: undefined: tls.SignatureScheme
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:317: undefined: tls.ECDSAWithP256AndSHA256
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:318: undefined: tls.ECDSAWithP384AndSHA384
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:318: undefined: tls.ECDSAWithP521AndSHA512
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:344: undefined: tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
GO/src/golang.org/x/crypto/acme/autocert/autocert.go:347: undefined: tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
```
I can't find any executable. The instructions on https://gotify.net/docs/dev-environment aren't enough for me to successfully compile and run the server.

**Comment by nkestrel** [comment]
I was able to build it for a Raspberry Pi 1 Model B but had difficulty with the web UI. The last supported version of Node.js on ARMv6 was too old, so I had to build on desktop but cross-compiling resulted in a GLIBC version mismatch.

The easiest approach for me was to build the web UI on desktop and then copy the resulting `packed-packr.go` to the RPi and finish the build there. `packr` needed to have been run on the RPi build environment beforehand so that it could see the imported package.

Here's the resulting binary for version 2.0.20-2-g3454dcd: [gotify-linux-arm-6.zip](https://github.com/gotify/server/files/5723656/gotify-linux-arm-6.zip)

Performance on RPi depends heavily on SD card speed, a slow one can result in a 4-8 second push delay.
